### PR TITLE
Allow the KeyStoreAccessor to work on API < 21

### PR DESCRIPTION
**Pull request by AdamBJohnsonx**
Noticed a bug filed against the KeyStoreAccessor involving instrumented tests. Since this is used in upcoming broker features, I thought I'd better fix it.  Currently, this enables it to work on API down to 18.

Testing-wise, I created a bunch of emulators and forced on desugaring, then ran the KeyStoreAccessor tests in each of them:

30
23
22
21
20
19
18
17 <- fails.

Since this is intended for use mainly by the broker code and OneAuth, at least for now, this should be acceptable.

### CODE REVIEW

**Review by iambmelt — APPROVED** [review]
I don't see any issues here

File: common/src/androidTest/java/com/microsoft/identity/common/internal/platform/KeyStoreAccessorTests.java [hunk 1]
```diff
@@ -83,20 +86,42 @@ public void testBasicFunctionalityDecryptDoesSomething() throws Exception {
...
+        boolean exception = false;
+        try {
+            KeyAccessor accessor = KeyStoreAccessor.newInstance(SymmetricCipher.AES_GCM_NONE_HMACSHA256, false);
+            byte[] in = new byte[1024];
+            RANDOM.nextBytes(in);
+            byte[] out = accessor.sign(in);
+        } catch (UnsupportedOperationException e) {
+            exception = true;
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+                throw e;
```
Comment (shahzaibj) [inline common/src/androidTest/java/com/microsoft/identity/common/internal/platform/KeyStoreAccessorTests.java:100]:
```suggestion
                throw new AssertionError(e);
```

File: common/src/androidTest/java/com/microsoft/identity/common/internal/platform/KeyStoreAccessorTests.java [hunk 2]
```diff
@@ -83,20 +86,42 @@ public void testBasicFunctionalityDecryptDoesSomething() throws Exception {
...
+        boolean exception = false;
+        try {
+            KeyAccessor accessor = KeyStoreAccessor.newInstance(SymmetricCipher.AES_GCM_NONE_HMACSHA256, false);
+            byte[] in = new byte[1024];
+            RANDOM.nextBytes(in);
+            accessor.verify(in, in);
+        } catch (UnsupportedOperationException e) {
+            exception = true;
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+                throw e;
```
Comment (shahzaibj) [inline common/src/androidTest/java/com/microsoft/identity/common/internal/platform/KeyStoreAccessorTests.java:119]:
```suggestion
                throw new AssertionError(e);;
```

File: common/src/androidTest/java/com/microsoft/identity/common/internal/platform/KeyStoreAccessorTests.java [hunk 3]
```diff
@@ -83,20 +86,42 @@ public void testBasicFunctionalityDecryptDoesSomething() throws Exception {
...
-        KeyAccessor accessor = KeyStoreAccessor.newInstance(SymmetricCipher.AES_GCM_NONE_HMACSHA256, false);
-        byte[] in = new byte[1024];
-        RANDOM.nextBytes(in);
-        byte[] out = accessor.sign(in);
+        boolean exception = false;
+        try {
+            KeyAccessor accessor = KeyStoreAccessor.newInstance(SymmetricCipher.AES_GCM_NONE_HMACSHA256, false);
+            byte[] in = new byte[1024];
+            RANDOM.nextBytes(in);
+            byte[] out = accessor.sign(in);
```
Comment (shahzaibj) [inline common/src/androidTest/java/com/microsoft/identity/common/internal/platform/KeyStoreAccessorTests.java:96]: nit: final

File: common/src/androidTest/java/com/microsoft/identity/common/internal/platform/KeyStoreAccessorTests.java [hunk 4]
```diff
@@ -83,20 +86,42 @@ public void testBasicFunctionalityDecryptDoesSomething() throws Exception {
...
-        KeyAccessor accessor = KeyStoreAccessor.newInstance(SymmetricCipher.AES_GCM_NONE_HMACSHA256, false);
-        byte[] in = new byte[1024];
-        RANDOM.nextBytes(in);
-        accessor.verify(in, in);
+        boolean exception = false;
+        try {
+            KeyAccessor accessor = KeyStoreAccessor.newInstance(SymmetricCipher.AES_GCM_NONE_HMACSHA256, false);
+            byte[] in = new byte[1024];
+            RANDOM.nextBytes(in);
+            accessor.verify(in, in);
```
Comment (shahzaibj) [inline common/src/androidTest/java/com/microsoft/identity/common/internal/platform/KeyStoreAccessorTests.java:115]: nit: final

**Review by shahzaibj — APPROVED** [review]
left some minor suggestions

File: common/src/androidTest/java/com/microsoft/identity/common/internal/platform/KeyStoreAccessorTests.java [hunk 5]
```diff
@@ -83,20 +86,42 @@ public void testBasicFunctionalityDecryptDoesSomething() throws Exception {
...
+    @Test
     public void testBasicFunctionalityUnsupportedSign() throws Exception {
-        KeyAccessor accessor = KeyStoreAccessor.newInstance(SymmetricCipher.AES_GCM_NONE_HMACSHA256, false);
-        byte[] in = new byte[1024];
-        RANDOM.nextBytes(in);
-        byte[] out = accessor.sign(in);
+        boolean exception = false;
+        try {
+            KeyAccessor accessor = KeyStoreAccessor.newInstance(SymmetricCipher.AES_GCM_NONE_HMACSHA256, false);
+            byte[] in = new byte[1024];
```
Comment (AdamBJohnsonx) [inline common/src/androidTest/java/com/microsoft/identity/common/internal/platform/KeyStoreAccessorTests.java:94]:
```suggestion
            final byte[] in = new byte[1024];
```

File: common/src/androidTest/java/com/microsoft/identity/common/internal/platform/KeyStoreAccessorTests.java (hunk 3 above)
Comment (AdamBJohnsonx) [inline common/src/androidTest/java/com/microsoft/identity/common/internal/platform/KeyStoreAccessorTests.java:96]:
```suggestion
            final byte[] out = accessor.sign(in);
```

File: common/src/androidTest/java/com/microsoft/identity/common/internal/platform/KeyStoreAccessorTests.java [hunk 6]
```diff
@@ -83,20 +86,42 @@ public void testBasicFunctionalityDecryptDoesSomething() throws Exception {
...
-    @Test(expected = UnsupportedOperationException.class)
+    @Test
     public void testBasicFunctionalityUnsupportedSign() throws Exception {
-        KeyAccessor accessor = KeyStoreAccessor.newInstance(SymmetricCipher.AES_GCM_NONE_HMACSHA256, false);
-        byte[] in = new byte[1024];
-        RANDOM.nextBytes(in);
-        byte[] out = accessor.sign(in);
+        boolean exception = false;
+        try {
+            KeyAccessor accessor = KeyStoreAccessor.newInstance(SymmetricCipher.AES_GCM_NONE_HMACSHA256, false);
```
Comment (AdamBJohnsonx) [inline common/src/androidTest/java/com/microsoft/identity/common/internal/platform/KeyStoreAccessorTests.java:93]:
```suggestion
            final KeyAccessor accessor = KeyStoreAccessor.newInstance(SymmetricCipher.AES_GCM_NONE_HMACSHA256, false);
```

File: common/src/androidTest/java/com/microsoft/identity/common/internal/platform/KeyStoreAccessorTests.java (hunk 4 above)
Comment (AdamBJohnsonx) [inline common/src/androidTest/java/com/microsoft/identity/common/internal/platform/KeyStoreAccessorTests.java:115]:
```suggestion
            final KeyAccessor accessor = KeyStoreAccessor.newInstance(SymmetricCipher.AES_GCM_NONE_HMACSHA256, false);
            final byte[] in = new byte[1024];
            RANDOM.nextBytes(in);
            accessor.verify(in, in);
```

**Pull request merged**